myia/compile/backends/python/implementations.py

```diff
@@ -0,0 +1,70 @@
+"""Global implementations useful for compiled modules."""
+from collections import Counter
+
+from ovld import ovld
+
+
+@ovld
+def myia_iter(obj: range):
+    return obj
+
+
+@ovld
+def myia_iter(obj: tuple):  # noqa: F811
+    return obj
+
+
+@ovld
+def myia_hasnext(obj: range):
+    return obj.start < obj.stop
+
+
+@ovld
+def myia_hasnext(obj: tuple):  # noqa: F811
+    return bool(obj)
+
+
+@ovld
+def myia_next(obj: range):
+    return obj.start, range(obj.start + obj.step, obj.stop, obj.step)
+
+
+@ovld
+def myia_next(obj: tuple):  # noqa: F811
+    return obj[0], obj[1:]
+
+
+def typeof(obj: object):
+    """Implementation for apply `typeof`."""
+    return type(obj)
+
+
+class MakeHandle:
+    """Helper class for apply `make_handle`."""
+
+    def __init__(self):
+        """Initialize."""
+        self.counter = Counter()
+
+    def __call__(self, object_type):
+        """Generate an handle for given object type."""
+        name = object_type.__name__
+        self.counter.update([name])
+        count = self.counter[name]
+        return f"{name}_h{count}"
```

Comment:
Seems a bit indirect. Why not just have a Handle class?

```suggestion
_counter = Counter()

class Handle:
    """Implementation of `make_handle`."""

    def __init__(self, object_type):
        """Generate an handle for given object type."""
        name = object_type.__name__
        _counter.update([name])
        count = _counter[name]
        self.name = f"{name}_h{count}"
        
    def __str__(self):
        return self.name
```

Technically you probably don't need the counter since every instance of Handle is unique.